.github/haskell.yml

```diff
@@ -0,0 +1,69 @@
+name: Haskell CI
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: ubuntu-18.04
+
+    steps:
+    - uses: actions/checkout@v1
+
+    - name: Install Haskell build environment
+      run: |
+        sudo apt-get update
+        sudo apt-get -y install software-properties-common
+        sudo add-apt-repository ppa:hvr/ghc
+
+        sudo apt-get -y install locales curl sudo git
+        sudo apt-get -y install g++ libstdc++6
+        sudo apt-get -y install libsnappy1v5 libtinfo-dev libsnappy-dev jq
+        sudo apt-get -y install zlib1g zlib1g-dev
+        sudo apt-get -y install libz3-dev
+        sudo apt-get -y install libpq5 libpq-dev
+        sudo apt-get -y install alex-3.1.7 cabal-install-3.0 happy-1.19.5
+        sudo apt-get -y install hpack hspec-discover libyaml-0-2
+        sudo apt-get -y install ghc-8.8.1
+        sudo apt-get -y install zip
+        sudo apt-get -y install wget
+
+        sudo apt-get -y remove --purge software-properties-common
+        sudo apt-get -y autoremove
+
+        sudo ln -s /opt/cabal/bin/cabal /usr/local/bin/cabal
+        sudo ln -s /opt/ghc/bin/ghc /usr/local/bin/ghc
+        sudo ln -s /opt/ghc/bin/ghc-pkg /usr/local/bin/ghc-pkg
+        cabal --version
+        ghc --version
+
+    - name: Install cabal-cache tool
+      run: |
+        CACHE_TOOL_TAG=$(curl -s -X GET https://api.github.com/repos/haskell-works/cabal-cache/releases/latest | jq -rc .tag_name)
+        echo "Downloading: https://github.com/haskell-works/cabal-cache/releases/download/${CACHE_TOOL_TAG}/cabal-cache_x86_64_linux.tar.gz"
+        curl -Ls https://github.com/haskell-works/cabal-cache/releases/download/${CACHE_TOOL_TAG}/cabal-cache_x86_64_linux.tar.gz | tar -xvz -C /tmp/
+        sudo cp /tmp/cabal-cache /usr/local/bin/cabal-cache
+        cabal-cache version
+
+    - name: Check cabal file
+      run: cabal check
+
+    - name: Cabal update
+      run: cabal v2-update
+
+    - name: Cabal Configure
+      run: cabal v2-configure
+
+    - name: Restore cabal cache
+      run: |
+        cabal-cache sync-from-archive \
+          --threads       8                                                                             \
+          --archive-uri   http://hw-binary-cache-us-west-2-a.s3-website-us-west-2.amazonaws.com/archive
+
+    - name: Install dependencies
+      run: cabal v2-build --enable-tests --enable-benchmarks --write-ghc-environment-files=always --only-dependencies
+
+    - name: Build
+      run: cabal v2-build --enable-tests --enable-benchmarks --write-ghc-environment-files=always
```

Comment:
What is your opinion on using stack instead of cabal? e.g. the CI integrations as mentioned in https://docs.haskellstack.org/en/stable/faq/ ?

Reply:
It's much better to use `cabal` because Haskell libraries uploaded to Hackage should just work and `stackage` pulls from `hackage` anyway.

If you wanted to additionally support `stack` users you don't actually need `stack` CI.  You just need to add your package to `build-constraints.yaml` to get included in the nightlies and LTS:

https://github.com/commercialhaskell/stackage/blob/master/build-constraints.yaml

You'll get notifications from them when your package breaks on `stack` and can rely on that for maintenance purposes.